### PR TITLE
Mage_Customer - IDE improvements

### PR DESCRIPTION
**Pull request by sreichel**
- added/fixed missing DOCs
- changed @return to $this for methods and annotations
- changed magic methods on Varien_Objects to setData()/getData()
- flipped @var annotations

### CODE REVIEW

File: app/code/core/Mage/Customer/Block/Account/Dashboard/Newsletter.php [hunk 1]
```diff
@@ -34,9 +34,17 @@
 
 class Mage_Customer_Block_Account_Dashboard_Newsletter extends Mage_Core_Block_Template
 {
+    /**
+     * @var Mage_Newsletter_Model_Subscriber
+     */
+    public $_subscription;
```
Comment (kkrieger85) [inline app/code/core/Mage/Customer/Block/Account/Dashboard/Newsletter.php]: Variables with underscore are hint for private attributes.

See [PEAR Manual](https://pear.php.net/manual/en/standards.naming.php)

File: app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php [hunk 2]
```diff
@@ -696,10 +606,9 @@ public function parse()
                         if ($customer->getDefaultBilling()) {
                             $model->setDefaultBilling($customer->getDefaultBilling());
                         } else {
-                            $shippingAddress->save();
+                            $billingAddress->save();
```
Comment (kkrieger85) [inline app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php]: This is a fundemental code change. 

Even if this is right, it should not be in this commit.

File: app/code/core/Mage/Customer/Block/Account/Dashboard/Newsletter.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Customer/Block/Account/Dashboard/Newsletter.php]: This var wasnt set and public before. Make it protected/private could cause problem with code relying on this?

File: app/code/core/Mage/Customer/controllers/AccountController.php [hunk 3]
```diff
@@ -652,7 +653,7 @@ public function confirmAction()
     public function confirmationAction()
     {
         $customer = $this->_getModel('customer/customer');
-        if ($this->_getSession()->isLoggedIn()) {
+        if (!$customer instanceof Mage_Customer_Model_Customer || $this->_getSession()->isLoggedIn()) {
```
Comment (sreichel) [inline app/code/core/Mage/Customer/controllers/AccountController.php]: Revert .... add @var annotation?

File: app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php]: Will be a own PR.

**Comment by colinmollenhour** [comment]
Just an FYI on the off chance you aren't already aware of this feature, to separate these changes into separate commits one good way is to use `git add --interactive` and once you get the hang of it, it is pretty fast. It basically presents each "chunk" one at a time so you can stage parts of a file without staging other parts.

File: app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php]: Resolved in #701

File: app/code/core/Mage/Customer/Block/Account/Dashboard/Newsletter.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Customer/Block/Account/Dashboard/Newsletter.php]: Fixed in #701

**Pull request closed without merging**